src/Database/Esqueleto/Record.hs

```diff
@@ -951,23 +880,7 @@ sqlMaybeSelectColCountDec RecordInfo {..} = do
                   InfixT lhs ''(:&) ty
              in foldl' helper t1 rest
 
-  -- Roughly:
-  -- sqlSelectColCount _ = sqlSelectColCount (Proxy @($joinedTypes))
-  pure $
-    FunD
-      'sqlSelectColCount
-      [ Clause
-          [WildP]
-          ( NormalB $
-              AppE (VarE 'sqlSelectColCount) $
-                ParensE $
-                  AppTypeE
-                    (ConE 'Proxy)
-                    joinedTypes
-          )
-          -- `where` clause.
-          []
-      ]
+  [d| sqlSelectColCount _ = sqlSelectColCount (Proxy @($(pure joinedTypes))) |]
```

Comment:
Huh - there are two mentions of `sqlSelectColCount`. This is one. Is it posible that the `[d| ... |]` here is doing `newName` instead of using the name in scope?

Reply:
I think you are right. This is quite odd behaviour (but also the type of issues I was hoping to find with this PR). I'll make a GHC ticket.

Strangely as welll
`[d| $(varP 'foo) = ...|]` is fine but as soon as I try to add arguments, it fails to parse, so I moved those into a lambda

Reply:
GHC issue: https://gitlab.haskell.org/ghc/ghc/-/issues/25435